torch/csrc/distributed/autograd/context/dist_autograd_context.h

```diff
@@ -40,9 +42,23 @@ class TORCH_API DistAutogradContext {
   DistAutogradContext(DistAutogradContext&&) = delete;
   DistAutogradContext& operator=(DistAutogradContext&&) = delete;
 
+  static_assert(
+      std::numeric_limits<rpc::worker_id_t>::max() <=
+          std::numeric_limits<int16_t>::max(),
+      "Range of possible workerIDs exceeds int16_t limit. This could cause "
+      "memory issues at runtime. Consider changing known_worker_set_type from "
+      "std::bitset to std::unordered_map");
+  using known_worker_set_type =
+      std::bitset<std::numeric_limits<rpc::worker_id_t>::max()>;
+  known_worker_set_type getKnownWorkerIds() const;
+  void addKnownWorkerID(const rpc::WorkerId& workerId);
+
  private:
   const int64_t context_id_;
 
+  // Bitset containing known worker IDs, used in cleaning up autograd context.
```

Comment:
Add some more details here about how we discover these worker ids. Basically, for each send function that we encounter in the autograd context, we store the destination here.